File: Code Forces/E85/c.py
def explode(i, hit):
	if(i > n):
		i = 1
	if new_monsters[i][0] > 0:
		new_monsters[i][0] -= hit

		if(new_monsters[i][0] <= 0):
			explode(i+1, new_monsters[i][1])
		else:
			return
	return

for _ in range(int(input())):
	n = int(input())
	monsters = {}
	for i in range(n):
		a, b = map(int, input().split())

		monsters[i+1] = [a, b]

	mons = sorted(monsters.items(), key = lambda k: k[1])

	bullets = 0
	new_monsters = {}
	for i , k in mons:
		new_monsters[i] = k
	
	for i, h in new_monsters.items():
		if h[0] > 0:
			bullets += h[0]
			explode(i+1, h[1])

	print(bullets)

File: Code Forces/E85/c.cpp
#include <bits/stdc++.h>
using namespace std;
int t, n;
vector<pair<int, pair<int, int>>> monsters;

bool comp(const pair<int, pair<int, int>>& lhs, const pair<int, pair<int, int>>& rhs){
  if(lhs.second.first < rhs.second.first || lhs.second.first > rhs.second.first)
  	return lhs.second.first < rhs.second.first;
  return lhs.second.second > rhs.second.second;
}

void explode(int i, int hit){
	if(i > n)
		i = 1;

	if(monsters[])
}

int main(){
	cin>>t;
	while(t--){
		cin>>n;
		for (int i = 1; i <= n; ++i){
			int a, b;
			cin>>a>>b;
			monsters.push_back({i, {a, b}});
		}

		sort(monsters.begin(), monsters.end(), comp);

		int bullets = 0;
		for (int i = 0; i < n; ++i){
			if(monsters[i].second.first > 0){
				bullets += monsters[i].second.first;
				explode(monsters[i].first + 1, monsters[i].second.second);
			}
		}
	}



	return 0;
}